main.py

```diff
@@ -163,9 +164,30 @@ def handle_errors(self, event):
         The event is sent only if the error is related to a request made
         by flow_manager.
         """
-        xid = event.content["message"].header.xid.value
-        error_type = event.content["message"].error_type
-        error_code = event.content["message"].code
+        message = event.content["message"]
+        connection = event.source
+        switch = connection.switch
+
+        error_data = message.data.pack()
+        # Get the packet responsible for the error
+        error_packet = connection.protocol.unpack(error_data)
+
+        if is_ofbac_bad_out_port(message.code):
+
+            for action in error_packet.actions:
+                try:
+                    iface = switch.get_interface_by_port_no(action.port.value)
+                except AttributeError:
+                    iface = switch.get_interface_by_port_no(action.port)
+
+                # Check interface to drop packets forwarded to it
+                if iface:
+                    version = connection.protocol.version
+                    iface.config = get_port_config_for_version(version)
+
+        xid = message.header.xid.value
+        error_type = message.error_type
+        error_code = message.code
```

Comment:
My suggestion is to do the more simple decoding first, as below.
I'd also add a comment above the "if is_ofbac_bad..." explaining what it does.

```suggestion
        message = event.content["message"]

        connection = event.source
        switch = connection.switch

        xid = message.header.xid.value
        error_type = message.error_type
        error_code = message.code       
        error_data = message.data.pack()

        # Get the packet responsible for the error
        error_packet = connection.protocol.unpack(error_data)

        if is_ofbac_bad_out_port(message.code):
            for action in error_packet.actions:
                try:
                    iface = switch.get_interface_by_port_no(action.port.value)
                except AttributeError:
                    iface = switch.get_interface_by_port_no(action.port)

                # Check interface to drop packets forwarded to it
                if iface:
                    version = connection.protocol.version
                    iface.config = get_port_config_for_version(version)
```